airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -256,6 +275,110 @@ public static ServerRunnable getServer(final ServerFactory apiFactory, final Con
+              sourceDef.getName());
+          final SynchronousResponse<ConnectorSpecification> getSpecJob = schedulerClient
+              .createGetSpecJob(sourceDef.getDockerRepository() + ":" + sourceDef.getDockerImageTag());
+          if (getSpecJob.isSuccess()) {
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Spec for Source Definition {} was successfully retrieved. Writing to the db...",
+                sourceDef.getName());
+            final StandardSourceDefinition updatedDef = Jsons.clone(sourceDef).withSpec(getSpecJob.getOutput());
+            configRepository.writeStandardSourceDefinition(updatedDef);
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Spec for Source Definition {} was successfully written to the db record.",
+                sourceDef.getName());
+          } else {
+            final LogRead logRead = jobConverter.getLogRead(getSpecJob.getMetadata().getLogPath());
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Failed to retrieve spec for Source Definition {}. Logs: {}",
+                sourceDef.getName(),
+                logRead.toString());
+            throw new RuntimeException(String.format(
+                "Failed to retrieve spec for Source Definition %s. Logs: %s",
+                sourceDef.getName(),
+                logRead.toString()));
+          }
+        }
+      } catch (final Exception e) {
+        trackSpecBackfillFailure(trackingClient, configRepository, sourceDef.getDockerRepository(), sourceDef.getDockerImageTag(), e);
+      }
+    }
+
+    for (final StandardDestinationDefinition destDef : configRepository.listStandardDestinationDefinitions()) {
+      try {
+        if (destDef.getSpec() == null) {
+          LOGGER.info(
+              "migrateAllDefinitionsToContainSpec - Destination Definition {} does not have a spec. Attempting to retrieve spec...",
+              destDef.getName());
+          final SynchronousResponse<ConnectorSpecification> getSpecJob = schedulerClient
+              .createGetSpecJob(destDef.getDockerRepository() + ":" + destDef.getDockerImageTag());
+          if (getSpecJob.isSuccess()) {
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Spec for Destination Definition {} was successfully retrieved. Writing to the db...",
+                destDef.getName());
+            final StandardDestinationDefinition updatedDef = Jsons.clone(destDef).withSpec(getSpecJob.getOutput());
+            configRepository.writeStandardDestinationDefinition(updatedDef);
+            LOGGER.info(
+                "migrateAllDefinitionsToContainSpec - Spec for Destination Definition {} was successfully written to the db record.",
+                destDef.getName());
+          } else {
+            final LogRead logRead = jobConverter.getLogRead(getSpecJob.getMetadata().getLogPath());
```

Comment:
This ended up working great, I tested it out and it doesn't seem to be _too_ much info to me, just a stacktrace of what went wrong in the get spec job. Here is an example:
`airbyte-server     | 2021-11-04 19:29:34 INFO i.a.s.ServerApp(migrateAllDefinitionsToContainSpec):316 - {workspace_app_root=/tmp/workspace/server/logs} - migrateAllDefinitionsToContainSpec - Failed to retrieve spec for Source Definition mk556 test 2. Logs: class LogRead {
airbyte-server     |     logLines: [2021-11-04 19:29:31 INFO () TemporalAttemptExecution(get):116 - Executing worker wrapper. Airbyte version: dev, 2021-11-04 19:29:31 INFO () LineGobbler(voidCall):82 - Checking if mk556/airbyte-custom-connectors:0.1.2 exists..., 2021-11-04 19:29:31 INFO () LineGobbler(voidCall):82 - mk556/airbyte-custom-connectors:0.1.2 not found locally. Attempting to pull the image..., 2021-11-04 19:29:34 INFO () LineGobbler(voidCall):82 - Image does not exist., 2021-11-04 19:29:34 INFO () TemporalAttemptExecution(lambda$getWorkerThread$2):170 - Completing future exceptionally..., io.airbyte.workers.WorkerException: Error while getting spec from image mk556/airbyte-custom-connectors:0.1.2, 	at io.airbyte.workers.DefaultGetSpecWorker.run(DefaultGetSpecWorker.java:74) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.DefaultGetSpecWorker.run(DefaultGetSpecWorker.java:23) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:167) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at java.lang.Thread.run(Thread.java:832) [?:?], Caused by: io.airbyte.workers.WorkerException: Could not find image: mk556/airbyte-custom-connectors:0.1.2, 	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:77) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.process.AirbyteIntegrationLauncher.spec(AirbyteIntegrationLauncher.java:52) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.DefaultGetSpecWorker.run(DefaultGetSpecWorker.java:44) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	... 3 more, 2021-11-04 19:29:34 INFO () TemporalAttemptExecution(get):137 - Stopping cancellation check scheduling..., 2021-11-04 19:29:34 WARN () POJOActivityTaskHandler$POJOActivityImplementation(execute):243 - Activity failure. ActivityId=8f0a08f5-04c1-3112-8854-658ba254ff71, activityType=Run, attempt=1, java.util.concurrent.ExecutionException: io.airbyte.workers.WorkerException: Error while getting spec from image mk556/airbyte-custom-connectors:0.1.2, 	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395) ~[?:?], 	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2063) ~[?:?], 	at io.airbyte.workers.temporal.TemporalAttemptExecution.get(TemporalAttemptExecution.java:135) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.temporal.spec.SpecActivityImpl.run(SpecActivityImpl.java:66) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?], 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?], 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?], 	at java.lang.reflect.Method.invoke(Method.java:564) ~[?:?], 	at io.temporal.internal.sync.POJOActivityTaskHandler$POJOActivityInboundCallsInterceptor.execute(POJOActivityTaskHandler.java:277) ~[temporal-sdk-1.0.4.jar:?], 	at io.temporal.internal.sync.POJOActivityTaskHandler$POJOActivityImplementation.execute(POJOActivityTaskHandler.java:216) ~[temporal-sdk-1.0.4.jar:?], 	at io.temporal.internal.sync.POJOActivityTaskHandler.handle(POJOActivityTaskHandler.java:181) ~[temporal-sdk-1.0.4.jar:?], 	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:192) ~[temporal-sdk-1.0.4.jar:?], 	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:154) ~[temporal-sdk-1.0.4.jar:?], 	at io.temporal.internal.worker.PollTaskExecutor.lambda$process$0(PollTaskExecutor.java:73) ~[temporal-sdk-1.0.4.jar:?], 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?], 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?], 	at java.lang.Thread.run(Thread.java:832) [?:?], Caused by: io.airbyte.workers.WorkerException: Error while getting spec from image mk556/airbyte-custom-connectors:0.1.2, 	at io.airbyte.workers.DefaultGetSpecWorker.run(DefaultGetSpecWorker.java:74) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.DefaultGetSpecWorker.run(DefaultGetSpecWorker.java:23) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:167) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	... 1 more, Caused by: io.airbyte.workers.WorkerException: Could not find image: mk556/airbyte-custom-connectors:0.1.2, 	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:77) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.process.AirbyteIntegrationLauncher.spec(AirbyteIntegrationLauncher.java:52) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.DefaultGetSpecWorker.run(DefaultGetSpecWorker.java:44) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.DefaultGetSpecWorker.run(DefaultGetSpecWorker.java:23) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:167) ~[io.airbyte-airbyte-workers-0.30.28-alpha.jar:?], 	... 1 more]`